File: api/predict.js
export default function handler(req, res) {
  // Enable CORS
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).json({ status: 'OK' });
  }

  if (req.method === 'POST') {
    try {
      const { image } = req.body;
      
      if (!image) {
        return res.status(400).json({ error: 'No image provided' });
      }

      // Mock prediction
      const has_tumor = Math.random() > 0.5;
      const confidence = Math.random() * 0.3 + 0.7; // 70-100%

      const result = {
        prediction: has_tumor ? 'Tumor detected' : 'No tumor detected',
        has_tumor: has_tumor,
        confidence: confidence
      };

      return res.status(200).json(result);
    } catch (error) {
      return res.status(500).json({ error: error.message });
    }
  }

  // Health check
  return res.status(200).json({ status: 'Scanix AI API is running' });
}

File: api/predict.py
from http.server import BaseHTTPRequestHandler
import json
import random

class handler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
    
    def do_POST(self):
        try:
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length)
            data = json.loads(post_data.decode('utf-8'))
            
            if not data or 'image' not in data:
                self.send_error_response({'error': 'No image data provided'}, 400)
                return
            
            # Mock prediction for Vercel (ML libraries too heavy)
            has_tumor = random.choice([True, False])
            confidence = random.uniform(0.75, 0.95)
            
            result = {
                'prediction': 'Tumor detected' if has_tumor else 'No tumor detected',
                'has_tumor': has_tumor,
                'confidence': confidence,
                'model_used': 'mock_vercel'
            }
            
            self.send_success_response(result)
            
        except Exception as e:
            self.send_error_response({
                'error': f'Prediction failed: {str(e)}',
                'details': 'Please check your image format and try again'
            }, 500)
    
    def send_success_response(self, data):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Content-Type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps(data).encode())
    
    def send_error_response(self, data, status_code):
        self.send_response(status_code)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Content-Type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps(data).encode())